tests/e2e/csi_cns_telemetry_vc_reboot.go

```diff
@@ -0,0 +1,374 @@
+		4. Wait for PVC's status to become bound.
+		5. Create pod using PVC.
+		6. Wait for Disk to be attached to the node.
+		7. Reboot VC.
+		8. Update cluster-distribution value and create PVC after VC reboots.
+		9. Wait for PVC to be bound.
+		10. Create pod using the PVC.
+		12. Wait for Disk to be attached to the node.
+		13. Delete pod and Wait for Volume Disk to be detached from the Node.
+		14. Delete PVC, PV and Storage Class.
+	*/
+	ginkgo.It("[csi-block-vanilla] [csi-file-vanilla] verify volume operations after vc reboots", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		var storageclass *storagev1.StorageClass
+		var pvclaim *v1.PersistentVolumeClaim
+		var fullSyncWaitTime int
+		var err error
+
+		// Read full-sync value
+		if os.Getenv(envFullSyncWaitTime) != "" {
+			fullSyncWaitTime, err = strconv.Atoi(os.Getenv(envFullSyncWaitTime))
+			framework.Logf("Full-Sync interval time value is = %v", fullSyncWaitTime)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+
+		ginkgo.By("Rebooting VC")
+		vcAddress := e2eVSphere.Config.Global.VCenterHostname + ":" + sshdPort
+		err = invokeVCenterReboot(vcAddress)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		err = waitForHostToBeUp(e2eVSphere.Config.Global.VCenterHostname)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		ginkgo.By(fmt.Sprintf("Waiting for %v for host to come up fully", vcRebootWaitTime))
+		time.Sleep(time.Duration(vcRebootWaitTime) * time.Second)
+		ginkgo.By("Done with reboot")
+
+		//After reboot
+		bootstrap()
+
+		// Sleep for a short while for the VC to shutdown, before invoking PVC creation and cluster-distribution value
+		time.Sleep(pollTimeoutShort)
+
+		ginkgo.By("Set cluster-distribution value after the VC reboot in progress")
+		// Set Cluster-distribution while the VC reboot in progress
+		setClusterDistribution(ctx, client, vanillaClusterDistributionWithSpecialChar)
+
+		defer func() {
+			//Reset the cluster distribution value to default value "CSI-Vanilla"
+			setClusterDistribution(ctx, client, vanillaClusterDistribution)
+		}()
+
+		ginkgo.By("Creating Storage Class and PVC")
+		// decide which test setup is available to run
+		if vanillaCluster {
+			ginkgo.By("CNS_TEST: Running for Vanilla setup")
+			storageclass, pvclaim, err = createPVCAndStorageClass(client, namespace, nil, nil, "", nil, "", false, "")
+		}
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		defer func() {
+			err := client.StorageV1().StorageClasses().Delete(ctx, storageclass.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		defer func() {
+			err = fpv.DeletePersistentVolumeClaim(client, pvclaim.Name, namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Waiting for claim to be in bound phase")
+		pvc, err := fpv.WaitForPVClaimBoundPhase(client, []*v1.PersistentVolumeClaim{pvclaim}, framework.ClaimProvisionTimeout)
```

Comment:
`WaitForPVClaimBoundPhase` block can be moved before the defer func block

Reply:
Fixed